### PR TITLE
TSK-385 created Rest endpoints for domains

### PR DESCRIPTION
**Pull request by LarsLeo**

### CODE REVIEW

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/ClassificationController.java [hunk 1]
```diff
@@ -1,31 +1,23 @@
...
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.PutMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
-
-import pro.taskana.Classification;
-import pro.taskana.ClassificationService;
-import pro.taskana.ClassificationSummary;
+import org.springframework.web.bind.annotation.*;
```
Comment (MiguelMartinRojas) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/ClassificationController.java]: Do not use wildcard if you already know which resources are you going to import

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/ClassificationController.java [hunk 2]
```diff
@@ -1,31 +1,23 @@
 package pro.taskana.rest;
```
Comment (MiguelMartinRojas) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/ClassificationController.java]: Do not use initial commit as commit comment, type something which describes the commit

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/ClassificationController.java [hunk 3]
```diff
@@ -76,6 +68,15 @@
         }
     }
 
+    @GetMapping(path = "/domains")
+    @Transactional(readOnly = true, rollbackFor = Exception.class)
+    public ResponseEntity<List<String>> getDomains() {
+        List<String> export = new ArrayList<>();
+        ClassificationQuery classificationQuery = classificationService.createClassificationQuery();
```
Comment (MiguelMartinRojas) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/ClassificationController.java]: I am not sure if createClassificationQuery can trigger exceptions, make sure it won't trigger them. In other case use try catch.

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/ClassificationController.java [hunk 4]
```diff
@@ -76,6 +68,15 @@
         }
     }
 
+    @GetMapping(path = "/domains")
+    @Transactional(readOnly = true, rollbackFor = Exception.class)
+    public ResponseEntity<List<String>> getDomains() {
+        List<String> export = new ArrayList<>();
```
Comment (MiguelMartinRojas) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/ClassificationController.java]: use a intuitive naming for export. something like domains or domainList

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/WorkbasketDefinitionController.java [hunk 5]
```diff
@@ -1,40 +1,30 @@
...
 import org.springframework.transaction.annotation.Transactional;
 import org.springframework.transaction.interceptor.TransactionInterceptor;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RequestBody;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.bind.annotation.RestController;
-
+import org.springframework.web.bind.annotation.*;
```
Comment (MiguelMartinRojas) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/WorkbasketDefinitionController.java]: I you do not change a file don't commit it. This file doesn't have any relevant change.

**Comment by tituslabs** [comment]
for the time being this approach should work for the admin workspace due to the appropriate role. but please add a todo comment to adapt this with the planed feature to configure domain names.

File: rest/taskana-rest-spring/src/main/java/pro/taskana/rest/ClassificationController.java [hunk 6]
```diff
@@ -76,6 +79,15 @@
         }
     }
 
+    @GetMapping(path = "/domains")
+    @Transactional(readOnly = true, rollbackFor = Exception.class)
```
Comment (TitusLabs) [inline rest/taskana-rest-spring/src/main/java/pro/taskana/rest/ClassificationController.java]: for read only, you can omit the rollbackFor parameter

**Pull request merged**